src/ui/password-input.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import { type Control, type FieldValues,type Path,useController } from 'react-hook-form';
+import { StyleSheet,View } from 'react-native';
+import { Eye, EyeOff } from 'react-native-feather';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+
+import { black } from './colors';
+import { Input } from './input';
+import { Text } from './text';
+
+interface PasswordInputProps<T extends FieldValues> {
+  control: Control<T>;
+  name: Path<T>;
+  label: string;
+  rules?: object;
+  testID?: string;
+}
+
+export function PasswordInput<T extends FieldValues>({
+  control,
+  name,
+  label,
+  rules,
+  testID,
+}: PasswordInputProps<T>) {
+ 
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const { field, fieldState } = useController({ control, name, rules });
+  
+  return (
+    <View
+      style={{
+        display: 'flex',
+       width: '100%',
+      }}>
+      <Text className= 'text-grey-100 text-lg dark:text-neutral-100'>
+          {label}
+      </Text>
+
+      <View style={passwordStyles.container}>
+        <Input
+          onChangeText={field.onChange}
+          value={(field.value as string) || ''}
+          className="pl-2"
+            style={{
+              height: 40,
+              width: 320,
+              flex: 1,
+              alignSelf: 'center',
+            }}
```

Comment:
This styles attributes can be set through the className property. So we always use tailwind except on certain situations like a variable style where the styleProp is required. 

Also in case stylesheets is preferred it's always better to handle the stylesheet on a separate styles file to avoid cluttering the code :)

Reply:
Sure I'll take a look in the futures code improvements